torch_xla/core/dynamo_bridge.py

```diff
@@ -308,12 +315,55 @@ def extract_internal(xla_model: torch.fx.GraphModule):
   # should be removed to avoid extra computation executed and in place updates op
   # mistakenlly update the input tensors.
   torch_xla._XLAC._clear_pending_irs(str(xm.xla_device()))
+  return (xla_args_sharding_spec, args_and_out, graph_hash,
+          arg_index_to_need_update_index, none_remover, graph_input_matcher,
+          dumb_return_handler, xla_args_need_update)
+
+
+def extract_internal(xla_model: torch.fx.GraphModule):
+  (xla_args_sharding_spec, args_and_out, graph_hash,
+   arg_index_to_need_update_index, none_remover, graph_input_matcher,
+   dumb_return_handler, xla_args_need_update) = extract_graph_helper(xla_model)
+  skip_checking_input_sharding_threashold = xu.getenv_as(
+      'XLA_DYNAMO_INPUT_SHARDING_CHECK_THRESHOLD', int, 5)
```

Comment:
Why would we want this flag? And how to use this flag properly?

Reply:
getting the shardingspec for all input tensor is not free, it has some speed implication for the inference(~ 5% - 10%) on 7B. The idea of this flag is that if input sharding is the same for `XLA_DYNAMO_INPUT_SHARDING_CHECK_THRESHOLD` times, we will assume everything is OK and stop checking the input shardings. It is more or less a hack.

Reply:
The core idea is that user should not change the input sharding. We are mainly solving the problem for compiler overwrite the sharding. Compiler usually overwrites sharding for the 1st run and then sharding will be the same.

Reply:
Then I guess this flag should be off by default? We should ensure program correctness first than provide some risky hacks for users to tune the performance?

Reply:
> The core idea is that user should not change the input sharding. We are mainly solving the problem for compiler overwrite the sharding. Compiler usually overwrites sharding for the 1st run and then sharding will be the same.

Okay, it makes sense then.